### PR TITLE
Codebreaker console for RubyGarage

### PR DESCRIPTION
**Pull request by Turzhanskyi**
Task details are [here](https://docs.google.com/document/d/1VW3Mk1W-pGkq0FadPih689_k971Zy8inzk6UCPHDLzs/edit)

### CODE REVIEW

**Review by n1zx1 — APPROVED** [review]

**Review by zelenyid — APPROVED** [review]
Looks good

File: lib/services/game_process.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    end
+
+    def request_of_hint
+      return message(:'errors.message.no_hints') unless game.hints_available?
+
+      puts "#{I18n.t(:'play.hint')}: #{game.hint}"
+    end
+
+    def describe_difficulty_levels
+      Codebreaker::Constants::LEVELS.each_key do |name|
```
Comment (nico-saj) [inline lib/services/game_process.rb]: `Codebreaker::Constants::LEVELS.each do |name, difficulty|`?

File: lib/services/game_process.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        puts compare(guess.value)
+      elsif guess.value == I18n.t(:'play.hint')
+        request_of_hint
+      else
+        message(:'errors.message.wrong_command')
+      end
+    end
+
+    def process_answer_menu(answer)
+      send(MAIN_MENU_COMMANDS[answer.to_sym])
```
Comment (nico-saj) [inline lib/services/game_process.rb]: What will happen if answer is not included in `MAIN_MENU_COMMANDS`?

File: lib/services/game_process.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module CodebreakerConsole
+  module GameProcess
+    MAIN_MENU_COMMANDS = { start: :start, rules: :rules, stats: :stats, exit: :exit }.freeze
```
Comment (nico-saj) [inline lib/services/game_process.rb]: Do u really need doubling commands in key and value? It's better to use array

**Review by nico-saj — CHANGES_REQUESTED** [review]

File: lib/services/game_process.rb (hunk 2 above)
Comment (Turzhanskyi) [reply lib/services/game_process.rb]: We got message "'You have passed unexpected command. Please choose one from listed commands.'"
after it "Enter command to continue: [start|rules|stats|exit]"

puts ```message(:'errors.message.unexpected_command')``` from
```
module CodebreakerConsole::class Console

def main_menu
      message(:main_menu)
      answer = user_enter
      return process_answer_menu(answer) if MAIN_MENU_COMMANDS.include?(answer)

      message(:'errors.message.unexpected_command')
      main_menu
end
````

File: lib/services/game_process.rb (hunk 1 above)
Comment (Turzhanskyi) [reply lib/services/game_process.rb]: fixed

File: lib/services/game_process.rb (hunk 3 above)
Comment (Turzhanskyi) [reply lib/services/game_process.rb]: fixed

**Comment by Turzhanskyi** [comment]
@nico-saj  fixed. Please, review)

**Review by nico-saj — APPROVED** [review]

**Pull request merged**